File: src/models/Asset.js
const mongoose = require('mongoose');

const assetSchema = new mongoose.Schema({
  lender: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  name: {
    type: String,
    required: true
  },
  description: {
    type: String,
    required: true
  },
  category: {
    type: String,
    required: true
  },
  condition: {
    type: String
  },
  originalPrice: {
    type: Number,
    required: false
  },
  purchaseYear: {
    type: Number,
    required: false
  },
  itemConditionRate: {
    type: Number,
    required: false // 0-100%
  },
  depositCalculationMode: {
    type: String,
    enum: ['fixed', 'auto'],
    default: 'fixed'
  },
  pricePerDay: {
    type: Number,
    required: true
  },
  depositAmount: {
    type: Number,
    required: true
  },
  status: {
    type: String,
    enum: ['draft', 'pending_approval', 'verified', 'rented', 'rejected', 'unavailable', 'maintenance', 'deleted'],
    default: 'pending_approval'
  },
  images: [{
    type: String
  }],
  videos: [{
    type: String
  }],
  serialNumber: {
    type: String,
    default: ''
  },
  invoiceImage: {
    type: String,
    default: ''
  },
  warrantyCardImage: {
    type: String,
    default: ''
  },
  badges: [{
    type: String,
    default: []
  }],
  aiAntiFraudStatus: {
    isCopied: { type: Boolean, default: false },
    reason: { type: String, default: '' },
    scannedAt: { type: Date }
  },
  inspectionChecklist: {
    type: mongoose.Schema.Types.Mixed,
    default: {}
  },
  specs: {
    type: Map,
    of: String,
    default: {}
  },
  location: {
    lat: { type: Number, required: true },
    lng: { type: Number, required: true },
    addressString: { type: String, default: '' }
  },
  verifiedBy: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User'
  },
  verificationNotes: {
    type: String
  },
  blockedDates: [{
    startDate: { type: Date, required: true },
    endDate: { type: Date, required: true },
    reason: { type: String, enum: ['manual', 'rented'], default: 'manual' }
  }]
}, { timestamps: true });

module.exports = mongoose.model('Asset', assetSchema);

File: mobile/lib/models/asset.dart
class Asset {
  final String id;
  final String name;
  final String description;
  final String category;
  final String? condition;
  final double pricePerDay;
  final double depositAmount;
  final String status;
  final List<String> images;
  final double? lat;
  final double? lng;
  final String? addressString;
  final String? lenderId;
  final String? lenderName;
  final String? lenderAvatar;
  final double? distance;
  final double? rating;
  final int? ratingCount;
  final List<String> badges;

  Asset({
    required this.id,
    required this.name,
    required this.description,
    required this.category,
    this.condition,
    required this.pricePerDay,
    required this.depositAmount,
    this.status = 'verified',
    this.images = const [],
    this.lat,
    this.lng,
    this.addressString,
    this.lenderId,
    this.lenderName,
    this.lenderAvatar,
    this.distance,
    this.rating,
    this.ratingCount,
    this.badges = const [],
  });

  factory Asset.fromJson(Map<String, dynamic> json) {
    final location = json['location'] as Map<String, dynamic>?;
    final lender = json['lender'] as Map<String, dynamic>?;
    return Asset(
      id: json['_id'] ?? json['id'] ?? '',
      name: json['name'] ?? '',
      description: json['description'] ?? '',
      category: json['category'] ?? '',
      condition: json['condition'],
      pricePerDay: _toDouble(json['pricePerDay']),
      depositAmount: _toDouble(json['depositAmount']),
      status: json['status'] ?? 'verified',
      images: _toStringList(json['images']),
      lat: location != null ? _toDouble(location['lat']) : null,
      lng: location != null ? _toDouble(location['lng']) : null,
      addressString: location?['addressString'],
      lenderId: lender?['_id']?.toString(),
      lenderName: lender?['name'],
      lenderAvatar: lender?['avatar'],
      distance: json['distance'] != null
          ? (json['distance'] as num).toDouble()
          : null,
      rating: json['avgRating'] != null
          ? (json['avgRating'] as num).toDouble()
          : (json['rating'] != null ? (json['rating'] as num).toDouble() : null),
      ratingCount: json['ratingCount'] != null ? (json['ratingCount'] as num).toInt() : null,
      badges: _toStringList(json['badges']),
    );
  }

  static double _toDouble(dynamic v) =>
      (v is num) ? v.toDouble() : 0.0;

  static List<String> _toStringList(dynamic v) {
    if (v is List) return v.map((e) => e.toString()).toList();
    return [];
  }
}
